src/main/java/perkpack/controllers/CardRestController.java

```diff
@@ -0,0 +1,185 @@
+                                         @RequestParam(value = "newDescription", required = false) String newDescription){
+        Card c = cardRepository.findOne(id);
+
+        if(c == null){
+            return ResponseEntity.badRequest().build();
+        }
+
+        if(!newName.isEmpty() || newName != null){
+            c.setName(newName);
+        }
+
+        if(!newDescription.isEmpty() || newDescription != null){
+            c.setDescription(newDescription);
+        }
+
+        Card card = cardRepository.save(c);
+
+        return ResponseEntity.ok().body(card);
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<Iterable<Card>> getCards(){
+        Iterable<Card> cards = cardRepository.findAll();
+        if(cards == null){
+            return ResponseEntity.badRequest().build();
+        }
+        return ResponseEntity.ok().body(cards);
+    }
+
+    @RequestMapping(value = "/{id}/addPerk/{perkName}", method = RequestMethod.PATCH)
+    public ResponseEntity<Card> addPerkToCard(@PathVariable("id") Long id,
+                                         @PathVariable(value = "perkName") String perkName){
+        Card c = cardRepository.findOne(id);
+        Perk perkToAdd = perkRepository.findByName(perkName);
+
+        if(c == null || perkToAdd == null){
+            return ResponseEntity.badRequest().build();
+        }
+
+        c.addPerk(perkToAdd);
+
+        Card card = cardRepository.save(c);
+
+        return ResponseEntity.ok().body(card);
+    }
+
+    @RequestMapping(value = "/{id}/removePerk/{perkName}", method = RequestMethod.PATCH)
+    public ResponseEntity<Card> removePerkFromCard(@PathVariable("id") Long id,
+                                                   @PathVariable(value = "perkName") String perkName){
+        Card c = cardRepository.findOne(id);
+
+        Perk perkToRemove = perkRepository.findByName(perkName);
+
+        if(c == null || perkToRemove == null){
+            return ResponseEntity.badRequest().build();
+        }
+
+        c.removePerk(perkToRemove);
+        Card card = cardRepository.save(c);
+
+        return ResponseEntity.ok().body(card);
+    }
+
+    @RequestMapping(value = "/{id}/addUser/{userID}", method = RequestMethod.PATCH)
+    public ResponseEntity<Card> addUserToCard(@PathVariable("id") Long id,
+                                              @PathVariable(value = "userID") Long userID){
+        Card c = cardRepository.findOne(id);
+        User userToAdd = userRepository.findOne(userID);
+
+        if(c == null || userToAdd == null){
+            return ResponseEntity.badRequest().build();
+        }
+
+        c.addUser(userToAdd);
+
+        Card card = cardRepository.save(c);
+
+        return ResponseEntity.ok().body(card);
+    }
+
+    /*
+    @RequestMapping(value = "/{id}/addUserToCardByEmail/{userEmail}", method = RequestMethod.PATCH)
+    public ResponseEntity<Card> addUserToCardByEmail(@PathVariable("id") Long id,
```

Comment:
I'm guessing that you can't include this functionality yet, because you're blocked by someone else?

Reply:
Not really. Apparently the endpoint cant be found when testing the function. I get a http 400 error which doesnt make sense. I figured ill just comment it out for now and work on it later

Reply:
Sounds good.